src/application/use-cases/DeleteUserUseCase.ts

```
@@ -0,0 +1,21 @@
+export class DeleteUserUseCase {
+  constructor(
+    private cavosWalletProvider: CavosWalletProvider,
+    private userRepository: UserRepository
+  ) { }
+
+  async execute(id: string, userId: string): Promise<{ message: string }> {
+    try {
+      await this.userRepository.deleteUserById(id);
+      await this.cavosWalletProvider.deleteUser(userId);
+    } catch (error) {
+      throw error;
```

Biome (1.9.4)

[error] The catch clause that only rethrows the original error is useless.

An unnecessary catch clause can be confusing.
Unsafe fix: Remove the try/catch clause.


(lint/complexity/noUselessCatch)